lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -344,31 +345,35 @@ inlineInputOnly(hw::HWModuleOp oldMod, hw::InstanceGraph &instanceGraph,
 
   // Check if it's ok to inline. We cannot inline the module if there exists a
   // declaration with an inner symbol referred by non-bind ops (e.g. hierpath).
+  auto oldModName = oldMod.getModuleNameAttr();
   for (auto port : oldMod.getPorts()) {
     if (port.sym) {
-      // Reject if the inner sym is a per-field symbol, or its inner ref is used
-      // by non-bind op.
-      if (!port.sym.getSymName() || port.sym.size() != 1 ||
-          innerRefUsedByNonBindOp.count(hw::InnerRefAttr::get(
-              oldMod.getModuleNameAttr(), port.sym.getSymName()))) {
-        oldMod.emitWarning() << "module " << oldMod.getModuleName()
-                             << " is an input only module but cannot "
-                                "be inlined because a signal "
-                             << port.name << " is referred by name";
-        return;
+      for (auto property : port.sym) {
+        auto innerRef = hw::InnerRefAttr::get(oldModName, property.getName());
+        if (innerRefUsedByNonBindOp.count(innerRef)) {
+          oldMod.emitWarning() << "module " << oldMod.getModuleName()
+                               << " is an input only module but cannot "
+                                  "be inlined because a signal "
+                               << port.name << " is referred by name";
+          return;
+        }
       }
     }
   }
-  for (auto &op : *oldMod.getBodyBlock()) {
-    if (auto innerSym = op.getAttrOfType<StringAttr>("inner_sym"))
-      if (innerRefUsedByNonBindOp.count(
-              hw::InnerRefAttr::get(oldMod.getModuleNameAttr(), innerSym))) {
-        op.emitWarning()
-            << "module " << oldMod.getModuleName()
-            << " is an input only module but cannot be inlined because signals "
-               "are referred by name";
-        return;
+
+  for (auto op : oldMod.getBodyBlock()->getOps<hw::InnerSymbolOpInterface>()) {
+    if (auto innerSym = op.getInnerSymAttr()) {
+      for (auto property : innerSym) {
+        auto innerRef = hw::InnerRefAttr::get(oldModName, property.getName());
+        if (innerRefUsedByNonBindOp.count(innerRef)) {
+          op.emitWarning() << "module " << oldMod.getModuleName()
+                           << " is an input only module but cannot be inlined "
+                              "because signals "
+                              "are referred by name";
```

Comment:
Nit: this line can be combined with the one above. I'm surprised clang-format didn't do that automatically.

Reply:
Yeah, I hate that it can't reflow them.  Drives me crazy :(.